main.go

```diff
@@ -91,25 +95,42 @@ var staticContents embed.FS
 func main() {
 	flag.Parse()
 
-	var tlsConfig *tls.Config
-	if flagRedisTLS != "" {
-		tlsConfig = &tls.Config{ServerName: flagRedisTLS}
+	var err error
+	var opts *redis.Options
+
+	if flagRedisURL != "" {
+		opts, err = redis.ParseURL(flagRedisURL)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		opts = &redis.Options{
+			Addr:      flagRedisAddr,
+			DB:        flagRedisDB,
+			Password:  flagRedisPassword,
+			TLSConfig: &tls.Config{},
+		}
+	}
+
+	if tls := opts.TLSConfig; tls != nil {
+		if tlsHost := flagRedisTLS; tlsHost != "" {
+			tls.ServerName = tlsHost
+		}
+
+		if flagRedisInsecureTLS {
+			tls.InsecureSkipVerify = true
+		}
```

Comment:
Since we have quite a bit of logic here, can you extract to a function? something like

```go
opts, err := getRedisOptionsFromFlags()
```

Reply:
Good suggestion 👍